### PR TITLE
Allow static linking of LLVM

### PR DESCRIPTION
**Pull request by winksaville**
Add LLVM_STATIC and update README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,6 +160,23 @@ Note that you can
...
+Typically you'll need to append `-DCMAKE_PREFIX_PATH=<path to llvm>` to the zig cmake line and if
+linking llvm statically also append `-DLLVM_STATIC=on`. For example, if llvm is
+installed at $HOME/local then for POSIX the cmake line `cmake ..` becomes:
+```
+cmake .. -DCMAKE_PREFIX_PATH=$HOME/local
+```
+And if statically linking LLVM:
+```
+cmake .. -DCMAKE_PREFIX_PATH=$HOME/local -DLLVM_STATIC=on
+```
```
Comment (bnoordhuis) [inline README.md]: Correct me if I'm wrong but this section only applies when `-DCMAKE_BUILD_TYPE=Debug` is also passed? Release builds seem to link statically against llvm.

**Comment by andrewrk** [comment]
Can you describe what problem you are trying to solve?

File: README.md (hunk 1 above)
Comment (winksaville) [reply README.md]: Yes

**Comment by winksaville** [comment]
@andrewrk, I think its a good idea that Debug and Release both be able to be statically linked. Although, as you can see [here](https://github.com/ziglang/zig/issues/67#issuecomment-483083332) that isn't always possible.

My vote would actually be to default to statically link LLVM and allow dynamic if desired/needed.

**Comment by emekoi** [comment]
i thought that was the status quo...

File: CMakeLists.txt [hunk 2]
```diff
@@ -47,6 +47,8 @@ option(ZIG_TEST_COVERAGE "Build Zig with test coverage instrumentation" OFF)
 # git log -p -- deps/lld
 option(ZIG_FORCE_EXTERNAL_LLD "If your system has the LLD patches use it instead of the embedded LLD" OFF)
 
+set(LLVM_STATIC off CACHE BOOL "Link LLVM statically")
```
Comment (andrewrk) [inline CMakeLists.txt]: Let's put this next to the corresponding `ZIG_STATIC` option, rename it to `ZIG_STATIC_LLVM`, and improve the documentation string. "Prefer linking against static LLVM libraries".

File: README.md [hunk 3]
```diff
@@ -160,6 +160,23 @@ Note that you can
 
 #### Instructions
 
+##### Optionally building LLVM
```
Comment (andrewrk) [inline README.md]: I'd like to omit this new section. The readme is a quick start guide. The supported method of discovering advanced build options is to read CMakeLists.txt. Teaching people how to install system dependencies is out of scope of the Zig readme. The supported method of installing Zig for people who do not know how to build dependencies from source or use their system package manager is to use the pre-built binaries from the download page.

**Comment by andrewrk** [comment]
Please feel free to re-open this PR with the requested changes.

**Pull request closed without merging**